src/t8_schemes/t8_new_feature/t8_subelements/t8_subelements_quad.h

```diff
@@ -0,0 +1,90 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_default_quad.h
+ * We use a p4est_quadrant_t object as storage for the T8 quadrant.
+ * To record if and if yes, how this quadrant is part of a 3D octant, we use
+ * the member pad8 for the surrounding toplevel dimension (2 or 3), pad16 for
+ * the direction of its normal relative to a toplevel octant (0, 1, or 2), and
+ * p.user_long for the p4est_qcoord_t coordinate in the normal direction.
+ */
+
+#ifndef T8_DEFAULT_QUAD_H
+#define T8_DEFAULT_QUAD_H
+
+#include <p4est.h>
+#include <t8_element.h>
+
+/** The structure holding a quadrilateral element in the default scheme.
+ * We make this definition public for interoperability of element classes.
+ * We might want to put this into a private, scheme-specific header file.
+ */
+
+//typedef p4est_quadrant_t t8_pquad_t;
+
+typedef struct
+{
+  p4est_quadrant_t    p4q;        /* p4est quadrant */
+  int                 dummy_is_subelement;      /* is element a subelement? */
+  int                 subelement_type;     /* which type of subelement to use? */
+  int                 num_subelement_types; /* # of parts for the specific kind of subelement */
```

Comment:
yes, you are right. 

num_subelement_types is deleted since it is an unnecessary information. 
num_subelement_ids is removed from the struct since it can always be determined by the get_number_of_subelement function

Reply:
The only important information we need are dummy_is_subelement, subelement_type and subelement_id